cedar-policy-core/src/ast/policy.rs

```diff
@@ -975,10 +975,27 @@ impl Annotations {
     pub fn iter(&self) -> impl Iterator<Item = (&AnyId, &Annotation)> {
         self.0.iter()
     }
+}
+
+/// Wraps the [`BTreeMap`]` into an opaque type so we can change it later if need be
+#[derive(Debug)]
+pub struct IntoIter(std::collections::btree_map::IntoIter<AnyId, Annotation>);
```

Comment:
What exactly are we getting by using this struct? Also, should probably give it a different name since we could conceivably want another `IntoIter` impl in the same module.